src/StoreApi/Routes/CartUpdateCustomer.php

```diff
@@ -75,74 +75,76 @@ public function get_args() {
-				'billing_email'       => isset( $request['billing_address'], $request['billing_address']['email'] ) ? $request['billing_address']['email'] : null,
-				'shipping_first_name' => isset( $shipping['first_name'] ) ? $shipping['first_name'] : null,
-				'shipping_last_name'  => isset( $shipping['last_name'] ) ? $shipping['last_name'] : null,
-				'shipping_company'    => isset( $shipping['company'] ) ? $shipping['company'] : null,
-				'shipping_address_1'  => isset( $shipping['address_1'] ) ? $shipping['address_1'] : null,
-				'shipping_address_2'  => isset( $shipping['address_2'] ) ? $shipping['address_2'] : null,
-				'shipping_city'       => isset( $shipping['city'] ) ? $shipping['city'] : null,
-				'shipping_state'      => isset( $shipping['state'] ) ? $shipping['state'] : null,
-				'shipping_postcode'   => isset( $shipping['postcode'] ) ? $shipping['postcode'] : null,
-				'shipping_country'    => isset( $shipping['country'] ) ? $shipping['country'] : null,
+				'billing_first_name'  => $billing['first_name'] ?? null,
+				'billing_last_name'   => $billing['last_name'] ?? null,
+				'billing_company'     => $billing['company'] ?? null,
+				'billing_address_1'   => $billing['address_1'] ?? null,
+				'billing_address_2'   => $billing['address_2'] ?? null,
+				'billing_city'        => $billing['city'] ?? null,
+				'billing_state'       => $billing['state'] ?? null,
+				'billing_postcode'    => $billing['postcode'] ?? null,
+				'billing_country'     => $billing['country'] ?? null,
+				'billing_phone'       => $billing['phone'] ?? null,
+				'billing_email'       => $billing['email'] ?? null,
+				'shipping_first_name' => $shipping['first_name'] ?? null,
+				'shipping_last_name'  => $shipping['last_name'] ?? null,
+				'shipping_company'    => $shipping['company'] ?? null,
+				'shipping_address_1'  => $shipping['address_1'] ?? null,
+				'shipping_address_2'  => $shipping['address_2'] ?? null,
+				'shipping_city'       => $shipping['city'] ?? null,
+				'shipping_state'      => $shipping['state'] ?? null,
+				'shipping_postcode'   => $shipping['postcode'] ?? null,
+				'shipping_country'    => $shipping['country'] ?? null,
+				'shipping_phone'      => $shipping['phone'] ?? null,
 			)
 		);
 
-		$shipping_phone_value = isset( $shipping['phone'] ) ? $shipping['phone'] : null;
-
-		// @todo Remove custom shipping_phone handling (requires WC 5.6+)
-		if ( is_callable( [ wc()->customer, 'set_shipping_phone' ] ) ) {
-			wc()->customer->set_shipping_phone( $shipping_phone_value );
-		} else {
-			wc()->customer->update_meta_data( 'shipping_phone', $shipping_phone_value );
-		}
+		/**
+		 * Fires when the Checkout Block/Store API updates a customer from the API request data.
+		 *
+		 * @param \WC_Customer $order Order object.
+		 * @param \WP_REST_Request $request Full details about the request.
+		 */
+		do_action( 'woocommerce_blocks_cart_update_customer_from_request', $customer, $request );
```

Comment:
This hook gives access to customer and request data. A 3rd party could grab the draft order manually if needed, but I'm not sure why you'd want to at this point.